### PR TITLE
Library managemet App changes

### PR DESCRIPTION
**Pull request by Alaguraj-S**

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  end
+  
+  def destroy
+    @book.destroy
+    respond_to do |format|
+      format.html {redirect_to books_url, notice: "Book was successfully destroyed." }
+    end
+  end
+
+    def search
```
Comment (Anuradha-Ravinarayanan) [inline app/controllers/books_controller.rb:63]: What is difference between search and book search? And if the method is not needed  please remove

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+class BooksController < ApplicationController
+  before_action :set_book, only: %i[ show edit update destroy ]
+  before_action :ensure_user_logged_in
+  before_action :ensure_admin_logged_in, only: [ :new, :create, :edit, :update, :destroy ]
+
+  def index
+  #  @books=Book.paginate(page: params[:page], per_page: 3)
+    @books=Book.search(params[:search]).paginate(page: params[:page], per_page: 4).order('title ASC')
```
Comment (Anuradha-Ravinarayanan) [inline app/controllers/books_controller.rb:8]: are we using search from here?

File: app/controllers/books_controller.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  
+  end
+
+  def new
+    @book = Book.new
+  end
+
+
+  def edit
+  end
```
Comment (Anuradha-Ravinarayanan) [inline app/controllers/books_controller.rb:30]: same if not needed please remove.

File: app/controllers/courses_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class CoursesController < ApplicationController
```
Comment (Anuradha-Ravinarayanan) [inline app/controllers/courses_controller.rb:1]: What is the use of this controler

File: app/controllers/userbooks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+class UserbooksController < ApplicationController
+    
+    def rent
+    BookUser.create(user_id: current_user.id, book_id: params[:book_id])
+    @book = Book.find(params[:book_id])
+    @book.Current_book -= 1;
```
Comment (Anuradha-Ravinarayanan) [inline app/controllers/userbooks_controller.rb:6]: please check the naming convention

File: app/controllers/userbooks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    @book.save
+    UserMailer.with(user: current_user,book: @book ).rent_mailer_alert.deliver_later
+    UserMailer.with(user: current_user,book: @book ).reminder_mailer_alert.deliver_later(wait_until: 5.minutes.from_now)
```
Comment (Anuradha-Ravinarayanan) [inline app/controllers/userbooks_controller.rb:9]: revisit the logic here

File: app/mailers/user_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+class UserMailer < ApplicationMailer
+    def login_alert
+    @user=params[:user]
```
Comment (Anuradha-Ravinarayanan) [inline app/mailers/user_mailer.rb:3]: is instance variable required here?

File: app/models/user.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    validates :email, presence: true
+    validates :name, presence: true
+
+    ADMIN = 'admin'.freeze
+    def is_admin?
+       role.eql?(ADMIN)
+    end
+
+    def self.search(search)
+        if search
```
Comment (Anuradha-Ravinarayanan) [inline app/models/user.rb:15]: Can it be done in controller itself instead of class method?

File: app/views/books/_book.html.erb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+ <table>
+ 
+        <thead>
+          <tr>
+              <th>Title</th>
+              <th>Author</th>
+              <th>publsihed date</th>
+              <% if current_user.role.eql?("admin")%>
+              <th> Book rented by</th>
```
Comment (Anuradha-Ravinarayanan) [inline app/views/books/_book.html.erb:11]: Please check the indentations

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    has_secure_password
+    has_many :book_users, dependent: :destroy
+    has_many :books, through: :book_users,dependent: :destroy
+
+    validates :email, presence: true
+    validates :name, presence: true
+
+    ADMIN = 'admin'.freeze
+    def is_admin?
+       role.eql?(ADMIN)
```
Comment (Anuradha-Ravinarayanan) [inline app/models/user.rb:11]: Please check the indentations

**Review by Anuradha-Ravinarayanan — CHANGES_REQUESTED** [review]
rc

File: app/controllers/courses_controller.rb (hunk 4 above)
Comment (Alaguraj-S) [reply app/controllers/courses_controller.rb:1]: For the index page of the app which request the user for signup/login

File: app/mailers/user_mailer.rb (hunk 7 above)
Comment (Alaguraj-S) [reply app/mailers/user_mailer.rb:3]: Yes, using the instance variable in app/views/user_mailer/login_alert.html.erb

File: app/controllers/userbooks_controller.rb (hunk 6 above)
Comment (Alaguraj-S) [reply app/controllers/userbooks_controller.rb:9]: Changed from "wait_until: **5.minutes**.from_now" to "wait_until: **7.days**.from_now"

**Pull request merged**